src/Platform/PlatformWindows.cpp

```diff
@@ -0,0 +1,290 @@
+        }
+    }
+
+    throw Exception("Falltergeist data files are not found!");
+}
+
+// this method looks for available dat files
+std::vector<std::string> *CrossPlatform::findFalloutDataFiles()
+{
+    if (_dataFiles)
+        return _dataFiles;
+
+    std::string path = CrossPlatform::findFalloutDataPath() + "*";
+    WIN32_FIND_DATA findData;
+    ZeroMemory(&findData, sizeof(findData));
+    HANDLE handle = FindFirstFileA(path.c_str(), &findData);
+    if (handle == INVALID_HANDLE_VALUE) {
+        throw Exception("Can't open data directory: " + CrossPlatform::findFalloutDataPath());
+    }
+
+    _dataFiles = new std::vector<std::string>(necessaryDatFiles);
+
+    do {
+        std::string filename(findData.cFileName);
+
+        if (filename.length() > 4) // exclude . and ..
+        {
+            std::string ext = filename.substr(filename.size() - 4, 4);
+            if (ext == ".dat")
+            {
+                if (filename.length() == 12 && filename.substr(0, 5) == "patch")
+                    _dataFiles->insert(_dataFiles->begin(), filename);
+            }
+        }
+
+    } while (FindNextFile(handle, &findData));
+
+    FindClose(handle);
+    return _dataFiles;
+}
+
+bool CrossPlatform::_createDirectory(const char *dir)
+{
+    DWORD attrs = GetFileAttributes(dir);
+
+    // Assume path exists
+    if (attrs != INVALID_FILE_ATTRIBUTES)
+    {
+        // Directory already exists
+        if (attrs & FILE_ATTRIBUTE_DIRECTORY) return false;
+
+        throw std::runtime_error("Path `" + std::string(dir) + "' already exists and is not a directory");
+    }
+    else
+    {
+        if (CreateDirectory(dir, NULL) != 0) return true;
+
+        DWORD errorId = GetLastError();
+
+        // FIXME: Use FormatMessage to get error string
+        throw std::runtime_error("CreateDirectory failed with code: " + std::to_string(errorId));
+    }
+    return false;
+}
+
+void CrossPlatform::createDirectory(std::string path)
+{
+    // Start index of 1 means that we skip root directory.
+    for (auto dirDelim = path.find_first_of('/', 1U); dirDelim != std::string::npos;
+         dirDelim = path.find_first_of('/', dirDelim + 1)) {
+        path[dirDelim] = '\0';
+        _createDirectory(path.c_str());
+        path[dirDelim] = '/';
+    }
+    _createDirectory(path.c_str());
+}
+
+std::string CrossPlatform::getConfigPath()
+{
+    char path[MAX_PATH];
+    SHGetFolderPath(nullptr, CSIDL_APPDATA | CSIDL_FLAG_CREATE, NULL, 0, path);
+    return std::string(path) + "/falltergeist";
+}
+
+std::vector<std::string> CrossPlatform::getDataPaths()
+{
+    std::vector<std::string> _dataPaths;
+    _dataPaths.push_back(getConfigPath());
+    return _dataPaths;
+}
+
+void CrossPlatform::getTime(TimeInfo * out) {
+    if (!out)
+        throw Exception("nullptr passed into getTime");
+
+    // Get ticks per second
+    LARGE_INTEGER Frequency;
+    if (QueryPerformanceFrequency(&Frequency) == FALSE)
```

Comment:
It will be better to call this function only once.

See https://msdn.microsoft.com/ru-ru/library/windows/desktop/ms644905(v=vs.85).aspx
`Retrieves the frequency of the performance counter. The frequency of the performance counter is fixed at system boot and is consistent across all processors. Therefore, the frequency need only be queried upon application initialization, and the result can be cached.`